# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1296101759 18000
#      Wed Jan 26 23:15:59 2011 -0500
# Node ID dd2af7b8dafe2f2eee74a78d849c32e4a6f0084b
# Parent  2ba90418168749ba00b9f107ca6c907be5453cb9
curl: simplify processing of output values

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-01-26  John W. Eaton  <jwe@octave.org>
+
+	* general/curl.m: Simplify processing of outptut values.
+
 2011-01-26  Olaf Till  <olaf.till@uni-jena.de>
 
 	* pkg/pkg.m: Export environment variables MKOCTFILE and
 	OCTAVE_CONFIG for configure and make, to configure also OCTAVE.
 
 2011-01-24  John W. Eaton  <jwe@octave.org>
 
 	* audio/lin2mu: Doc fix.
diff --git a/scripts/general/curl.m b/scripts/general/curl.m
--- a/scripts/general/curl.m
+++ b/scripts/general/curl.m
@@ -83,19 +83,17 @@ function varargout = curl (varargin)
          || (length (dy) != rows (varargin{fidx})))
       error ("curl: size of dx and dy must match the respective dimension of X and Y");
     endif
 
     dFx_dy = gradient (varargin{fidx}.', dy, dx).';
     dFy_dx = gradient (varargin{fidx + 1}, dx, dy);
     rot_z = dFy_dx - dFx_dy;
     av = rot_z / 2;
-    if (nargout == 0)
-      av
-    elseif (nargout == 1)
+    if (nargout == 0 || nargout == 1)
       varargout{1} = av;
     elseif (nargout == 2)
       varargout{1} = rot_z;
       varargout{2} = av;
     else
       error ("curl: number of output arguments must be 1 or 2");
     endif
 
@@ -116,28 +114,19 @@ function varargout = curl (varargin)
     rot_x = dFz_dy - dFy_dz;
     rot_y = dFx_dz - dFz_dx;
     rot_z = dFy_dx - dFx_dy;
     l = sqrt(varargin{fidx}.^2 + varargin{fidx + 1}.^2 + varargin{fidx + 2}.^2);
     av = (rot_x .* varargin{fidx} +
           rot_y .* varargin{fidx + 1} +
           rot_z .* varargin{fidx + 2}) ./ (2 * l);
 
-    if (nargout == 0)
-      varargout{1} = av;
-    elseif (nargout == 1)
+    if (nargout == 0 || nargout == 1)
       varargout{1} = av;
-    elseif (nargout == 2)
-      varargout{1} = rot_x;
-      varargout{2} = rot_y;
-    elseif (nargout == 3)
-      varargout{1} = rot_x;
-      varargout{2} = rot_y;
-      varargout{3} = rot_z;
-    elseif (nargout == 4)
+    else
       varargout{1} = rot_x;
       varargout{2} = rot_y;
       varargout{3} = rot_z;
       varargout{4} = av;
     else
       error ("curl: number of output arguments must be smaller than 5");
     endif
   endif
